models/ed/R/write.configs.ed.R

```diff
@@ -87,139 +87,27 @@ convert.samples.ED <- function(trait.samples){
 ##' and the name of the file to create
 ##' @name write.config.ED2
 ##' @title Write ED configuration files
-##' @param defaults list of defaults to process
-##' @param trait.samples vector of samples for a given trait
+##' @param trait.values Named list of trait values, with names corresponding to PFT
 ##' @param settings list of settings from pecan settings file
 ##' @param run.id id of run
+##' @param defaults list of defaults to process. Default=settings$constants
 ##' @return configuration file and ED2IN namelist for given run
 ##' @export
-##' @author David LeBauer, Shawn Serbin, Carl Davidson
+##' @author David LeBauer, Shawn Serbin, Carl Davidson, Alexey Shiklomanov
 ##-------------------------------------------------------------------------------------------------#
-write.config.ED2 <- function(defaults, trait.values, settings, run.id){
+write.config.ED2 <- function(trait.values, settings, run.id, defaults=settings$constants){
   
-  # find out where to write run/ouput
-  rundir <- file.path(settings$run$host$rundir, run.id)
-  outdir <- file.path(settings$run$host$outdir, run.id)
```

Comment:
Now that you move this line and the following block to write.config.jobsh.ED2 function, it can't find modeloutdir for writing the ED2IN file later:

```
ed2in.text <- gsub('@OUTDIR@', modeloutdir, ed2in.text)
```